src/main/java/eu/qped/java/checkers/design/DesignChecker.java

```diff
@@ -21,60 +22,206 @@
 @Builder
 public class DesignChecker {
 
-
+    /**
+     * answer as String
+     */
     private String answer;
 
-    private String targetProject;
+    /**
+     * answer as a multi-class project (given as path to package)
+     */
+    private String targetProjectOrFile;
+    private Compiler compiler;
+
+    private ArrayList<DesignFeedback> feedbacks;
+
+    private DesignSettingsReader designSettingsReader;
+
+    private QFDesignSettings qfDesignSettings;
+    private DesignSettings designSettings;
+
+    private final static String PATH_TO_CLASSFILES = "src/main/java/eu/qped/java/utils/compiler/compiledFiles";
 
     /**
      * is able to check one or multiple .class files
-     * for defined metrics ({@link Metric}).
-     * The output is printed on the console (plain or xml) or saved in {@link DesignCheckReport#getMetricsMap()} ).
+     * for defined metrics ({@link SaveMapResults.Metric}).
+     *
+     * @return the built {@link DesignCheckReport}
      */
     public DesignCheckReport check() {
+
         DesignCheckReport.DesignCheckReportBuilder resultBuilder = DesignCheckReport.builder();
-        Map<String, Map<Metric, Double>> metricsMap = new HashMap<>();
+        Map<String, Map<SaveMapResults.Metric, Double>> metricsMap = new HashMap<>();
+        DesignSettingsReader designSettingsReader = DesignSettingsReader.builder().qfDesignSettings(this.qfDesignSettings).build();
+        this.designSettings = designSettingsReader.readDesignSettings();
+
+        if (compiler == null) {
+            compiler = Compiler.builder().build();
+        }
+
+        boolean compileResult;
+        if (answer != null && !answer.equals("")) {
+            compileResult = compiler.compileFromString(answer);
+        } else {
+            compileResult = compiler.compileFromProject(targetProjectOrFile);
+        }
+
+        List<File> classFiles
+                = ExtractJavaFilesFromDirectory.builder().dirPath(PATH_TO_CLASSFILES).build().filesWithExtension("class");
+        String[] classFileNames = new String[classFiles.size()];
+        for (int i = 0; i < classFiles.size(); i++) {
```

Comment:
Foo[] array = list.toArray(new Foo[0]); oder? es geht einfacher als das ein List in einen Array zu kopieren